### ISSUE TITLE
`bun add` uses wrong cwd when running workspaces’ scripts

### ISSUE THREAD

**Issue by controversial** [oven-sh/bun#7517]
### What version of Bun is running?

1.0.15+b3bdf22eb

### What platform is your computer?

Darwin 23.1.0 arm64 arm

#### `package.json`:
```json
{
  "name": "parent",
  "workspaces": [
    "packages/package-1"
  ]
}
```

#### `packages/package-1/package.json`:
```json
{
  "name": "child",
  "scripts": {
    "postinstall": "echo PWD for postinstall is: $(pwd)"
  }
}
```

1. First, run `bun install` from the root. 
  Observe: “PWD for postinstall is: /path/to/my-project/packages/package-1”
2. Run `bun add lodash` from the root.
  Observe: “PWD for postinstall is: /path/to/my-project/packages/package-1”
3. Run `cd packages/package-1 && bun add lodash`
  Observe: **“PWD for postinstall is: /path/to/my-project”**

### What is the expected behavior?

The postinstall script for the workspace (`package-1`) should always be run with `package-1` as the cwd

### What do you see instead?

- If `bun install` or `bun add` is run from the *root*, the cwd for scripts is correct.
- If `bun add` is run from *within the package folder*, the cwd for scripts is **incorrect**, which can break scripts

### Additional information

NPM does not have this bug

This bug blocks me from using Bun’s package manager to add packages, as I have a postinstall script on some of my workspaces which is sensitive to its working directory.

**Comment by Jarred-Sumner** [comment]
sounds good, we'll fix

**Comment by controversial** [comment]
I believe this might be the same issue as #4891, but that issue is not very clearly described, and contains unrelated discussion, so I thought it would be worth it to create another

Thanks for the quick response!

**Comment by ftzi** [comment]
Edit3: Handling it here: https://github.com/oven-sh/bun/issues/7627

Seems to be related. My GitLab CI is outputting this when installing a lib:

![image](https://github.com/oven-sh/bun/assets/28177550/7fe8133b-fafc-4332-b8e7-e1ec12fa61b3)

It's postinstall command is `"postinstall": "node attribution.js"`. The package name is `@fortawesome/free-solid-svg-icons`. It's using cwd instead of the package's path.

Is it related to this? https://twitter.com/jarredsumner/status/1555008750120865792

Edit: And by trying to `bun install --ignore-scripts`, my project's `postinstall` doesn't run, which causes issues later on as it should convert `less` files to `css`.

Edit2: And, this postinstall package is listed here in the 500 packages: https://github.com/oven-sh/bun/blob/main/src/install/default-trusted-dependencies.txt (https://bun.sh/blog/bun-v1.0.17#bun-install-now-runs-lifecycle-scripts-for-the-top-500-npm-packages)

It's a bad situation/bug that involves different factors. @Jarred-Sumner can you take a look at this?

**Comment by dylan-conway** [comment]
Fixed in #11177, available in `v1.1.10`
